lib/OpenQA/Script/CloneJob.pm

```diff
@@ -90,21 +90,45 @@ sub clone_job_get_job ($jobid, $remote, $remote_url, $options) {
     return $job;
 }
 
+sub _job_setting_is ($job, $key, $expected_value) {
+    my $actual_value = $job->{settings}->{$key};
+    return $actual_value && $actual_value eq $expected_value;
+}
+
+sub _get_chained_parents ($job, $remote, $remote_url, $options, $parents = [], $parent_ids = {}) {
+    next if $parent_ids->{$job->{id}}++;
+    my @direct_parents = map { clone_job_get_job($_, $remote, $remote_url, $options) } @{$job->{parents}->{Chained}};
+    push @$parents, @direct_parents;
+    _get_chained_parents($_, $remote, $remote_url, $options, $parents, $parent_ids) for @direct_parents;
+    return $parents;
+}
+
+sub _is_asset_generated_by_cloned_jobs ($job, $parents, $file, $options) {
+    return 0 if ($options->{'skip-deps'} || $options->{'skip-chained-deps'}) && (scalar @$parents != 0);
+    for my $j (@$parents, $job) {
+        for my $setting (qw(PUBLISH_HDD_1 PUBLISH_PFLASH_VARS)) {
+            return 1 if _job_setting_is $j, $setting => $file;
+        }
+    }
+    return 0;
```

Comment:
Would be nice to combine grep+map here, something like the following maybe?

```suggestion
    grep { _job_setting_is $_, $setting => $file } …
```

Reply:
But the early return makes sense and the for loops are quite easy to read (compared to squashing too much logic into one line).